test/utils/set-deep.test.ts

```diff
@@ -228,6 +228,112 @@ describe('Unit | Utility | set deep', () => {
     });
   });
 
+  it('set on nested properties within an empty object', () => {
```

Comment:
I had to add this test, because my intuitive feelings about how arrays should work wasn't working.

idk if this is intended, or why it would be intended, but intuitively, I would have thoughts this would have been the case:

```ts
// ...
    expect(value).toEqual({
      top: {
        name: new Change('zoo'),
        foo: new Change({
          other: 'baz'
        })
      }
    });
```
because the `foo` is an entirely new object,

and how this correlates to arrays: 
instead of
```ts
it('sets when the array is initially empty', () => {
  const objA = {
    top: []
  };
  let value = setDeep(objA, 'top.0.name', new Change('zoo'));

  expect(value).toEqual({
    top: [{ name: new Change('zoo') }]
  });

  value = setDeep(value, 'top.0.foo.other', new Change('baz'));

  expect(value).toEqual({
    top: [
      {
        name: new Change('zoo'),
        foo: {
          other: new Change('baz')
        }
      }
    ]
  });
 });
```
maybe: 
```ts
it('sets when the array is initially empty', () => {
  const objA = {
    top: []
  };
  let value = setDeep(objA, 'top.0.name', new Change('zoo'));

  expect(value).toEqual({
    top: [new Change({ name: new Change('zoo') })]
  });

  value = setDeep(value, 'top.0.foo.other', new Change('baz'));

  // the big difference here -- the first entry remains an entire single Change
  expect(value).toEqual({
    top: [
      new Change({
        name: 'zoo',
        foo: {
          other: 'baz'
        }
      })
    ]
  });
});
```